app/controllers/application_controller.rb

```diff
@@ -88,6 +89,22 @@ def sublayout
 
   protected
 
+  def check_browser
+    request_format = request.format
+    return true if request_format.xml? || request_format.json?
+
+    if current_user && browser_not_modern?
+      logout_keeping_session!
+      flash.now[:error] = "The browser you are using doesn't seem to support the X-Frame-Options header. That means we can't protect you against Cross Frame Scripting and thus not guarantee the security of your session. Please upgrade your browser and sign in again."
+      redirect_to provider_admin_path
+    end
+  end
+
+  def browser_not_modern?
```

Comment:
I'd use the positive  `modern_browser?` here

Reply:
we would need to return true if a browser is unknown.

Reply:
taking into account that, I would prefer `modern_not_modern?`